substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/ResourceAttributesView.java

```diff
@@ -0,0 +1,117 @@
+package com.oracle.svm.core.jdk;
+
+import java.io.IOException;
+import java.nio.file.ReadOnlyFileSystemException;
+import java.nio.file.attribute.BasicFileAttributeView;
+import java.nio.file.attribute.FileAttributeView;
+import java.nio.file.attribute.FileTime;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ResourceAttributesView implements BasicFileAttributeView {
+
+    private enum AttributeID
+    {
+        size,
+        creationTime,
+        lastAccessTime,
+        lastModifiedTime,
+        isDirectory,
+        isRegularFile,
+        isSymbolicLink,
+        isOther,
+        fileKey,
+    }
+
+    private final ResourcePath path;
+    private final boolean isBasic;
+
+    public ResourceAttributesView(ResourcePath path, boolean isBasic) {
+        this.path = path;
+        this.isBasic = isBasic;
+    }
+
+    @SuppressWarnings("unchecked")
+    static <V extends FileAttributeView> V get(ResourcePath path, Class<V> type) {
+        if (type == null)
+            throw new NullPointerException();
+        if (type == BasicFileAttributeView.class)
```

Comment:
Always use {}. This style is deprecated for a long time.

Reply:
Done.